source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc

```diff
@@ -374,18 +387,37 @@ JsonTranscoderConfig::methodToRequestInfo(const MethodInfoSharedPtr& method_info
 
 ProtobufUtil::Status
 JsonTranscoderConfig::translateProtoMessageToJson(const Protobuf::Message& message,
-                                                  std::string* json_out) {
+                                                  std::string* json_out) const {
   return ProtobufUtil::BinaryToJsonString(
       type_helper_->Resolver(), Grpc::Common::typeUrl(message.GetDescriptor()->full_name()),
       message.SerializeAsString(), json_out, print_options_);
 }
 
 JsonTranscoderFilter::JsonTranscoderFilter(JsonTranscoderConfig& config) : config_(config) {}
 
+void JsonTranscoderFilter::initPerRouteConfig() {
+  if (!decoder_callbacks_->route() || !decoder_callbacks_->route()->routeEntry()) {
+    per_route_config_ = &config_;
+    return;
+  }
+
+  const std::string& name = HttpFilterNames::get().GrpcJsonTranscoder;
+  const auto* entry = decoder_callbacks_->route()->routeEntry();
+  const auto* route_local = entry->mostSpecificPerFilterConfigTyped<JsonTranscoderConfig>(name);
+
+  per_route_config_ = route_local ? route_local : &config_;
+}
+
 Http::FilterHeadersStatus JsonTranscoderFilter::decodeHeaders(Http::RequestHeaderMap& headers,
                                                               bool end_stream) {
+
+  initPerRouteConfig();
+  if (per_route_config_ == nullptr) {
```

Comment:
In what circumstance this would be true? from initPerRouteConfig() I don't think it can be null?

Reply:
yeah i think it's a leftover check from the times when `initPerRouteConfig` was not doing `per_route_config_ = &config` in the first if